src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/ReferralService.java

```diff
@@ -30,6 +34,19 @@
     private final OffenderDeletionBatchRepository batchRepository;
     private final RetentionService retentionService;
     private final ReferralResolutionService referralResolutionService;
+    private final DataComplianceEventPusher deletionGrantedEventPusher;
+
+    public void handleAdHocDeletion(final AdHocOffenderDeletion event) {
```

Comment:
Not sure you really need the type of event in the method name as this is determined by the type being passed in...meh